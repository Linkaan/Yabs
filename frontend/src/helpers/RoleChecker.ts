import UsersModule, { User } from '../store/modules/UsersModule';

export default class RoleChecker {

  private static checkPermission(user: User, required: number): boolean {
    return user && ((user.role & required) == required || (user.role > required));
  }
  
  static isPrincipal(): boolean {
    return this.checkPermission(UsersModule.currentUser, 32);
  }

  static isAdmin(): boolean {
    return this.checkPermission(UsersModule.currentUser, 16);
  }

  static isCaretaker(): boolean {
    return this.checkPermission(UsersModule.currentUser, 8);
  }

  static isTeacher(): boolean {
    return this.checkPermission(UsersModule.currentUser, 4);
  }

  static isStudentHealth(): boolean {
    return this.checkPermission(UsersModule.currentUser, 2);
  }

  static isStudent(): boolean {
    return UsersModule.currentUser && (UsersModule.currentUser.role & 1) == 1;
  }
}
