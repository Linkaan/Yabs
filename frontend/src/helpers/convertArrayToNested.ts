export default function convertList(payload: any[], typeOfId: string): object { //eslint-disable-line
  return payload.reduce((acc, item) => {
    acc[item[typeOfId]] = item;
    return acc;
  }, {});
}
