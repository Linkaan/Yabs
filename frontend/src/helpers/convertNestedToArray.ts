export default function convertList(payload: any, typeOfId: string): any[] { //eslint-disable-line
  const res = Object.keys(payload).map((id) => payload[id]);
  return res;
}
