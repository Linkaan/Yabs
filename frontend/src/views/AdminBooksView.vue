<template>
  <div>
    <h1 class="display-2">
      Books
    </h1>
    <div class="flex">
      <div style="margin-right: 50px">
        <BookFormComponent />
      </div>
      <div>
        <BookListComponent />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import BooksModule from '../store/modules/BooksModule';
import BookFormComponent from '@/components/BookFormComponent.vue';
import BookListComponent from '@/components/BookListComponent.vue';
import TitlesModule from '../store/modules/TitlesModule';
import { defineComponent } from '@vue/composition-api';

export default defineComponent({
  name: 'AdminBooksView',
  components: {
    BookFormComponent,
    BookListComponent
  },
  setup(): void{
    function created(): void {
      BooksModule.fetchAll();
      TitlesModule.fetchAll();
    }
    created();
  }
});
</script>

<style scoped>
.flex {
  display: flex;
}

.flex > div {
  flex: 1;
}
</style>
