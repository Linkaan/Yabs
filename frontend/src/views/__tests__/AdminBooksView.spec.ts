import AdminBooksView from '../AdminBooksView.vue';
import vue from 'vue';

import { Wrapper } from '@vue/test-utils';
import { shallowFactory } from '@/helpers/testFactoryHelpers';

jest.mock('@/store/modules/TitlesModule');
jest.mock('@/store/modules/BooksModule');


describe('AdminTitlesEditView.vue', () => {
  it('Displays the correct data in the form', async () => {
    const wrapper: Wrapper<vue> = shallowFactory(AdminBooksView);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
