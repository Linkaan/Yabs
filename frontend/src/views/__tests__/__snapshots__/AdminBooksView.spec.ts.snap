// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminTitlesEditView.vue Displays the correct data in the form 1`] = `
<div>
  <h1 class="display-2">
    BÃ¶cker
  </h1>
  <div class="flex">
    <div style="margin-right: 50px;">
      <bookformcomponent-stub></bookformcomponent-stub>
    </div>
    <div>
      <booklistcomponent-stub></booklistcomponent-stub>
    </div>
  </div>
</div>
`;
